src/views/SignUpView.vue

```diff
@@ -47,16 +90,105 @@
 </template>
 
 <script>
+import { mapActions } from 'pinia'
+import { useUserStore } from '../stores/user'
 import TheForm from '../components/shared/TheForm.vue'
 import TheInput from '../components/shared/TheInput.vue'
 import TheButton from '../components/shared/TheButton.vue'
+import ErrorToast from '../components/shared/ErrorToast.vue'
 
 export default {
   name: 'SignUpView',
   components: {
     TheForm,
     TheInput,
-    TheButton
+    TheButton,
+    ErrorToast
+  },
+  data() {
+    return {
+      userData: {
+        name: '',
+        passOne: '',
+        passTwo: '',
+        email: ''
+      },
+      errorMessages: {
+        nameError: '',
+        passOneError: '',
+        passTwoError: '',
+        emailError: '',
+        checkboxError: ''
+      },
+      isCheckboxChecked: false,
+      isFormSubmitted: false,
+      isToastShown: false
+    }
+  },
+  methods: {
+    ...mapActions(useUserStore, ['singUp']),
+    // eslint-disable-next-line consistent-return
+    checkLength(input, minChar, label) {
+      if (input === '') {
+        return `The "${label}" value is required.`
+      }
+
+      if (input !== '' && input.length < minChar) {
+        return `${label} must have at least ${minChar} characters.`
+      }
+
+      return ''
+    },
+    checkName() {
+      this.errorMessages.nameError = this.checkLength(this.userData.name, 5, 'User name')
+    },
+    checkPassword() {
+      this.errorMessages.passOneError = this.checkLength(this.userData.passOne, 8, 'Password')
+      this.errorMessages.passTwoError = this.checkLength(this.userData.passTwo, 0, 'Repeat password')
+
+      if (this.userData.passOne !== '' && this.userData.passTwo !== '' && this.userData.passOne !== this.userData.passTwo) {
+        this.errorMessages.passTwoError = "Passwords don't match."
+      }
+    },
+    checkEmail() {
+      this.errorMessages.emailError = this.checkLength(this.userData.email, 0, 'Email')
+      const emailConditions = /^(([^<>()[\]\.,;:\s@\"]+(\.[^<>()[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i
```

Comment:
Is that regex sth more than setting `type="email"` on the input field?

Reply:
Yep. This regular expression actually checks if the email put by a user is correct, since an email is suppose to be like `something@something.com`. So, this whole long expressions sets the correct email input to have some letters/words before the @ sign, then this @ sign, then some domain, then period and finally - this domain ending like .com, .pl, and so forth.

So, what I'm doing here is first, defining this correct email template and then, checking (using this test() method) if the data inputed by the user matches this template.

Reply:
but can this be achieved natively in the browser by setting `type="email"` to an input?

Reply:
If so consider using this: https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/checkValidity

Reply:
> but can this be achieved natively in the browser by setting `type="email"` to an input?

Don't think so. Setting `type="email"` to the input doesn't prevent a user from typing whatever he wants. [For example, take a look here.](https://stackoverflow.com/a/33204725) I mean, I wouldn't actually know any better, but, I don't think there is a way the browser could check the validity of the put email address.




> If so consider using this: https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/checkValidity

What are criteria for the validity in this thing? This article doesn't mention it at all, so I'm not able to state if this a solution for my issue.

Reply:
I think current state doesn't do it either, check my screenshot:
<img width="477" alt="image" src="https://user-images.githubusercontent.com/33238158/220487702-5e0d943c-f90d-41db-9ddf-eb4721d647d3.png">

I've made up a quick codepen, might be helpful:
https://codepen.io/redaxela/pen/GRXZNoG

Reply:
In your screenshot there are no errors because you didn't submit the form. But I checked it locally and my regexp in fact doesn't see any issue with this address, so you're right.

However, I tampered with your codepen a little and I noticed some interesting things.
First, this validity method returns true for the mentioned email, take a look.
![validity](https://user-images.githubusercontent.com/113309294/220773571-44fc7878-ddab-4287-8fde-cb88dfe9fe6f.PNG)
Second, it doesn't check the email as thoroughly as my regexp, watch this:
![validity_2](https://user-images.githubusercontent.com/113309294/220773716-7a655e8b-6cb5-4e33-af04-a9d9ae0f474f.PNG)
And finally, I still don't know about the criteria for validity (I can't find any info on that) but check what happened when I changed the input type for password:
![validity_3](https://user-images.githubusercontent.com/113309294/220773886-c6441085-7945-47f5-a118-9316c9dc90bd.PNG)

So, your codepen was actually helpful 😅

Reply:
Because it returns true for mostly valid addresses. There are many cases we should handle, but we're usually unaware of. Check https://stackoverflow.com/questions/201323/how-can-i-validate-an-email-address-using-a-regular-expression for more 🤓